MotionPhotoReader/app/src/main/java/com/google/android/libraries/motionphotoreader/MotionPhotoReaderUtils.java

```diff
@@ -0,0 +1,143 @@
+     */
+    public static void readFromVideoTrack(MediaExtractor extractor, MediaCodec decoder,
+                                          ByteBuffer inputBuffer, int bufferIndex) {
+        int sampleSize = extractor.readSampleData(inputBuffer, /* offset = */ 0);
+        if (sampleSize < 0) {
+            decoder.queueInputBuffer(
+                    bufferIndex,
+                    /* offset = */ 0,
+                    /* size = */ 0,
+                    /* presentationTimeUs = */ 0,
+                    MediaCodec.BUFFER_FLAG_END_OF_STREAM
+            );
+        } else {
+            decoder.queueInputBuffer(
+                    bufferIndex,
+                    /* offset = */ 0,
+                    sampleSize,
+                    extractor.getSampleTime(),
+                    /* flags = */ 0
+            );
+        }
+    }
+
+    /**
+     * Retrieve the information of the next available output buffer.
+     * @param availableOutputBuffers A blocking queue of Bundle objects containing information about
+     * free output buffers.
+     * @return the bundle containing the information of the next available output buffer, or null if
+     * the poll call results in a timeout.
+     */
+    public static Bundle getOutputBuffer(BlockingQueue<Bundle> availableOutputBuffers) {
+        Bundle bufferData = null;
+        try {
+            bufferData = availableOutputBuffers.poll(TIMEOUT_US, TimeUnit.MILLISECONDS);
+        } catch (InterruptedException e) {
+            Log.e(TAG, "No output buffer available", e);
+        }
+        return bufferData;
+    }
+
+    /**
+     * Retrieve the stabilization homographies from the motion track. If the frame has already been
+     * stabilized, then return an empty list.
+     * @param extractor The extractor connected to the track containing the stabilization data.
+     * @param inputBuffer A free input buffer for the extractor to read data into.
+     * @return a List of HomographyMatrix objects.
+     */
+    public static List<HomographyMatrix> getHomographies(MediaExtractor extractor,
+                                                         ByteBuffer inputBuffer,
+                                                         List<Float> prevHomographyDataList) {
+        List<HomographyMatrix> homographyList = new ArrayList<>();
+        boolean protocolBufferIsAvailable = true;
+        int sampleSize = extractor.readSampleData(inputBuffer, 0);
+        if (sampleSize >= 0) {
+            // Deserialize data
+            Stabilization.Data stabilizationData = null;
+            try {
+                stabilizationData = Stabilization.Data.parseFrom(inputBuffer);
+                if (stabilizationData == null) {
+                    Log.e(TAG, "Protocol buffer is null");
+                    protocolBufferIsAvailable = false;
+                }
+            } catch (InvalidProtocolBufferException e) {
+                Log.e(TAG, "Could not parse from protocol buffer");
```

Comment:
`protocolBufferIsAvailable` should be set to false too in that case too. Or better yet, it should be initialized to false and be only explicitly set to true in the code paths where you know it's valid.